### PR TITLE
Update expected figures to use matplotlib 3.5

### PR DESCRIPTION
**Pull request by danielhuppmann**
# Please confirm that this PR has done the following:

- [x] Tests Added
- ~Documentation Added~
- ~Name of contributors Added to AUTHORS.rst~
- [x] Description in RELEASE_NOTES.md Added

# Description of PR

Alerted by the nightly tests... It seems that matplotlib v3.5 changed the order of the legends in stackplot-figures with totals. This PR simply updates the test-figures to reflect the new behavior (alternative approach: change the way that the "total" is added to the legend).

In addition, the PR silences a warning in the plotting module and changes the readme and install-bat-file from `py.test` to `pytest` (because this seems to be the more common usage).

### CODE REVIEW

File: pyam/plotting.py [hunk 1]
```diff
@@ -437,7 +437,7 @@ def stack(
 
     def as_series(index, name):
         _idx = [i[0] for i in index]
-        return pd.Series([0] * len(index), index=_idx, name=name)
+        return pd.Series([0] * len(index), index=_idx, name=name, dtype="int")
```
Comment (gidden) [inline pyam/plotting.py:440]: Not clear why this is needed, can you let me know why?

Can this be sent through calling code? e.g., either with

1. move this `dtype` up as a kwarg or
2. add generic `**kwargs` which then passes through to this series call

**Review by gidden — COMMENTED** [review]
Thanks for fixing the nightlies! Small question in review.

File: pyam/plotting.py (hunk 1 above)
Comment (danielhuppmann) [reply pyam/plotting.py:440]: This is a function only used within this function (see just the following line) to select the rows where a timeseries crosses zero. Adding the dtype explicitly is done just to silence a warning in the log.

Comment (gidden) [reply pyam/plotting.py:440]: Gotcha - won't hold up. But I assume this will not work for non-integer times?

Comment (gidden) [reply pyam/plotting.py:440]: *facepalm* reread, and realize no issue here. sorry!

**Pull request merged**